builder.go

```diff
@@ -35,9 +35,16 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 
 	// Build the steps.
 	steps := []multistep.Step{
+		&StepSetup{
+			config: b.config,
+		},
 		&StepCloneVM{
 			config: b.config,
 		},
+		&StepCreateSnapshot{
```

Comment:
the goal of the parameter is making output VM suitable for linked clones.
If I set `cpus=2` and `create_snapshot=true` then the hardware customization is lost: the template has the setting, but child VMs do not inherit it.
Need to move snapshotting to a last action before final converting to a template.